### PR TITLE
Added orgtr

### PR DESCRIPTION
**Pull request by tfalders**
Partially addresses SWDEV-247072

### CODE REVIEW

File: rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.hpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  rocblas_get_stream(handle, &stream);
+
+  rocblas_stride strideW =
+      rocblas_stride(n - 1) * n / 2; // number of elements to copy
+  rocblas_int ldw = n - 1;
+  rocblas_int blocks = (n - 2) / BS + 1;
+
+  if (uplo == rocblas_fill_upper) {
+    // shift the householder vectors provided by gebrd as they come below the
+    // first subdiagonal
```
Comment (jzuniga-amd) [inline rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.hpp:93]: These comments are misplaced:
If upper, then the vectors are above the superdiagonal and need to be shifted left.
If lower, the vectors are below the subdiagonal and need to be shifted right

File: rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.hpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    // result
+    rocsolver_orgql_ungql_template<BATCHED, STRIDED, T>(
+        handle, n - 1, n - 1, n - 1, A, shiftA, lda, strideA, ipiv, strideP,
+        batch_count, scalars, work, Abyx_tmptr, trfact, workArr);
+  }
+
+  else {
+    // shift the householder vectors provided by gebrd as they come above the
+    // first superdiagonal
```
Comment (jzuniga-amd) [inline rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.hpp:113]: idem

**Review by jzuniga-amd — APPROVED** [review]
LGTM great job!!!
Just one minor correction.

**Comment by jzuniga-amd** [comment]
Good to go!

**Review by cgmb — APPROVED** [review]
A one-sentence description of what the copyshift functions do might be nice in a comment above the implementation of each. That said, this looks fairly straight-forward overall so I'm happy with it either way.

**Pull request merged**